jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ContextProvider.java

```diff
@@ -385,6 +385,10 @@ public ContextHandler createContextHandler(final App app) throws Exception
             // Handle a context XML file
             if (FileID.isXml(path))
             {
+                ClassLoader coreContextClassLoader = Environment.CORE.equals(environment) ? findCoreContextClassLoader(path) : null;
+                if (coreContextClassLoader != null)
+                    Thread.currentThread().setContextClassLoader(coreContextClassLoader);
```

Comment:
You should remove these lines from 408-409 below

Reply:
They are different: they set the classloader on the context, not on the thread.